### PR TITLE
EVM: Fix getBlock RPC and clean up RPC structs

### PR DESCRIPTION
**Pull request by sieniven**
## Summary

- Fix incorrect population of mixHash field to use block header mixHash value on RpcBlock and RpcBlockHeader
- Fix sealFields population to align with Parity-Ethereum (https://github.com/ethereum/EIPs/issues/95)
- Refactor RpcBlock to store header information using RpcBlockHeader struct with composition
- Fix RpcBlockHeader to implement from block header struct
- Refactor and clean up duplicate LogResult struct definition

## Implications

- Storage
  - [ ] Database reindex required
  - [ ] Database reindex optional
  - [ ] Database reindex not required
  - [x] None

- Consensus
  - [ ] Network upgrade required
  - [ ] Includes backward compatible changes
  - [ ] Includes consensus workarounds
  - [ ] Includes consensus refactors
  - [x] None

### CODE REVIEW

**Review by shohamc1 — DISMISSED** [review]

File: lib/ain-grpc/src/logs.rs [hunk 1]
```diff
@@ -7,29 +7,29 @@ use crate::block::BlockNumber;
...
 impl From<LogIndex> for LogResult {
     fn from(log: LogIndex) -> Self {
         Self {
+            address: log.address,
             block_hash: log.block_hash,
             block_number: log.block_number,
+            data: Bytes::from(log.data),
             log_index: log.log_index,
-            removed: log.removed,
+            removed: false,
```
Comment (prasannavl) [inline lib/ain-grpc/src/logs.rs:29]: Is this intentional?

Comment (sieniven) [reply lib/ain-grpc/src/logs.rs:29]: Ah definitely not. Thanks for the spot, fixed in 157240850d

File: lib/ain-grpc/src/block.rs [hunk 2]
```diff
@@ -10,80 +10,99 @@ use serde::{
...
+            mix_hash: header.mix_hash,
+            nonce: header.nonce,
+            number: header.number,
+            parent_hash: header.parent_hash,
+            receipts_root: header.receipts_root,
+            seal_fields: vec![
+                Bytes::from(header.mix_hash.as_fixed_bytes().to_vec()),
+                Bytes::from(header.nonce.as_fixed_bytes().to_vec()),
+            ],
+            sha3_uncles: H256::from_slice(&sha3::Keccak256::digest(
```
Comment (prasannavl) [inline lib/ain-grpc/src/block.rs:62]: We can pull these 0 hashes out, so we don't recompute

**Comment by prasannavl** [comment]
Blocked on https://github.com/DeFiCh/ain/pull/2653 for CI runs

File: lib/ain-grpc/src/block.rs (hunk 2 above)
Comment (prasannavl) [reply lib/ain-grpc/src/block.rs:62]: Will do in a separate PR along

**Pull request merged**